pkg/providers/vsphere/virtualmachine/extraconfig.go

```diff
@@ -13,11 +13,13 @@ import (
 )
 
 func IsPausedByAdmin(moVM mo.VirtualMachine) bool {
-	for i := range moVM.Config.ExtraConfig {
-		if o := moVM.Config.ExtraConfig[i].GetOptionValue(); o != nil {
-			if o.Key == vmopv1.PauseVMExtraConfigKey {
-				if value, ok := o.Value.(string); ok {
-					return strings.ToUpper(value) == constants.ExtraConfigTrue
+	if c := moVM.Config; c != nil {
+		for i := range c.ExtraConfig {
+			if o := c.ExtraConfig[i].GetOptionValue(); o != nil {
+				if o.Key == vmopv1.PauseVMExtraConfigKey {
+					if value, ok := o.Value.(string); ok {
+						return strings.ToUpper(value) == constants.ExtraConfigTrue
+					}
```

Comment:
Why not have a `return false` here? We shouldn't have dup keys.

Reply:
Good point! Didn't noticed it as I was just adding the nil check, updated.